language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionUtil.java

```diff
@@ -298,7 +298,7 @@ public static Map<TypeSymbol, String> getPossibleTypeSymbols(TypeSymbol typeDesc
                                 signature = "(" + newArrType.memberTypeDescriptor().signature() + ")[]";
                                 break;
                             default:
-                                signature = newArrType.signature();
+                                signature = entry.getValue() + "[]";
```

Comment:
We have to consider the following case as well. 
![extract_to_function](https://user-images.githubusercontent.com/35211477/207563912-0ccf922c-4e4c-44e3-a3cd-db6e3dd80659.gif)

```ballerina

type TestRecordType1 int;

type TestRecordType2 string;

public function TestFunc(string[] names, int x) returns string[]|error {
    (TestRecordType1|TestRecordType2)[] rec = [];
    foreach var item in names {
        TestRecordType1[] result = [1];
        rec.push(...result);
    }
}
```


We must use the `newArrType` instead of `entry`. Let's use FunctionGenerator.processModuleIdsInText() method to get the type name.

Reply:
Done